docs/contributing.md

```diff
@@ -49,6 +49,17 @@
   This strategy has worked well for Scala.js, and we intend to follow it too.
   * All contributions should be formatted with `./bin/scalafmt`.
 
+## Testing style
+
+  * In order to better reason about the evolution of the compiler,
+  we require that every commit in your pull request passes `bin/ci`.
```

Comment:
@densh We adopted this requirement from https://scala-native.readthedocs.io/en/latest/contrib/contributing.html#prepare-meaningful-commits, but I'm currently unsure how to check it automatically in Travis, since Travis seems to only run checks for the topmost commit. What do you do about that?

Reply:
I don’t know of any automated feature.  I do know some people who will go to the effort of manually pushing every commit one right after the other to get them all to run.  Seems hard to get people to honor that though.

Reply:
Are we doing squash merges?  If so, then every commit to master will end up compiling and still make bisecting possible.

Reply:
This is a good argument for squash merges. @densh, is this why you guys are using them?

Reply:
I think Travis has a build-every-commit setting, it just isn't enabled by default?

Reply:
Hmm I just check and can't find it. Could it be that it's limited to the enterprise edition?

Btw, @SethTisue, since you're here. How do you deal with this at scala/scala?

Reply:
All I know of is Travis has a “cancel existing build if a new commit comes in while building previous commit”.  This is not enabled by default and with that setting every push will be built, but not every commit.